src/smt/solver_engine.cpp

```diff
@@ -2248,6 +2248,14 @@ void SolverEngine::setOption(const std::string& key,
   }
   Trace("smt") << "SMT setOption(" << key << ", " << value << ")" << endl;
   options::set(getOptions(), key, value);
+  if (key == "stats-all")
+  {
+    d_env->d_statisticsRegistry.get()->setStatsAll(true);
+  }
+  else if (key == "stats-internal")
+  {
+    d_env->d_statisticsRegistry.get()->setStatsInternal(true);
+  }
```

Comment:
These shouldn't be set here. A better place would be `Env::finishInit`, where you can use `d_options.base.statisticsInternal` and `d_options.base.statisticsAll` to set the flags accordingly. `finishInit` is called when the options are finalized and the solver is fully initialized.

Reply:
Thank you!! I moved the code, and it is still printing statistics properly!